In [1]:
import numpy as np
from scipy import stats

# Data
physical_exercise = np.array([4.00, 2.67, 3.65, 2.11, 3.21, 3.60, 2.80])
no_physical_exercise = np.array([3.75, 2.74, 3.42, 1.67, 3.00, 3.25, 2.65])

# Paired t-test
t_statistic, p_value = stats.ttest_rel(physical_exercise, no_physical_exercise)

# Degrees of freedom
df = len(physical_exercise) - 1

# Critical value for a two-tailed test at 0.01 significance level
alpha = 0.01
t_critical = stats.t.ppf(1 - alpha / 2, df)

# Mean difference score
mean_difference = np.mean(physical_exercise - no_physical_exercise)

# Sample standard deviation
sample_std = np.std(physical_exercise - no_physical_exercise, ddof=1)

# Estimated standard error
std_error = sample_std / np.sqrt(len(physical_exercise))

# Cohen's d for paired samples
cohens_d = mean_difference / sample_std

# Output results
print("T-Statistic:", t_statistic)
print("Critical Value of t:", t_critical)
print("P-Value:", p_value)
print("Is the p-value less than 0.05?", p_value < 0.05)
print("Mean Difference Score:", mean_difference)
print("Sample Standard Deviation:", sample_std)
print("Estimated Standard Error:", std_error)
print("Cohen's d:", cohens_d)
print("Should we retain the null hypothesis?", p_value > alpha)

T-Statistic: 3.6644226528474397
Critical Value of t: 3.707428021324907
P-Value: 0.010523873876434144
Is the p-value less than 0.05? True
Mean Difference Score: 0.2228571428571428
Sample Standard Deviation: 0.16090517763725384
Estimated Standard Error: 0.06081644067012074
Cohen's d: 1.3850215768665572
Should we retain the null hypothesis? True


#### Markdown
$\overline{D}=0.22285714285714286$

$\Sigma{D}=1.5599999999999996$

$\Sigma{D^2}=0.503$

$SS_D=0.15534285714285728$

$S_D=0.16090517763725384$

$S_\overline{D}=0.06081644067012074$

$t=3.664422652847441$

$d=1.3850215768665575$

### Sample Sums of Squares
$SS_D=\Sigma{D}^2-\dfrac{(\Sigma{D})^2}{n}$

#### SAMPLE STANDARD DEVIATION, $S_D$

$S_D = \sqrt{\dfrac{SS_D}{n-1}}$

#### ESTIMATED STANDARD ERROR, $S_\overline{D}$

$S_\overline{D}=\dfrac{S_D}{\sqrt{n}}$

$t = \dfrac{\overline{D}-\mu_{D_{hyp}}}{S_\overline{D}}$

$d=\dfrac{\overline{D}}{S_D}$